draft-ietf-tls-esni.md

```diff
@@ -916,6 +916,32 @@ and session IDs presented by the server. These connections are only used to
 trigger retries, as described in {{rejected-ech}}. This may be implemented, for
 instance, by reporting a failed connection with a dedicated error code.
 
+
+### Impact of Retry on Future Connections
+
+Clients MAY use information learned from a rejected ECH for future
+connections to avoid repeatedly connecting to the same server and
+being forced to retry. However, they MUST handle ECH rejection for
+those connections as if it were a fresh connection, rather than
+enforcing the single retry limit from {{rejected-ech}}.
```

Comment:
This "however" was a little unclear to me at first.

I think that what you are saying is that when you save a retry config that worked for a future connection, you need to treat a retry that results as though you did not have a retry.  The result is that you will use the retry config and try again.

This needs to be part of the text that explains that you shouldn't accept a retry config if you just used one.  Otherwise, it's a little confusing.